File: RFriends-bungeecord/src/main/java/ga/justreddy/wiki/rfriends/command/commands/MessageCommand.java
package ga.justreddy.wiki.rfriends.command.commands;

import ga.justreddy.wiki.rfriends.command.ICommand;
import ga.justreddy.wiki.rfriends.utils.Utils;
import net.md_5.bungee.api.ProxyServer;
import net.md_5.bungee.api.connection.ProxiedPlayer;

public class MessageCommand implements ICommand {

  @Override
  public String name() {
    return "message";
  }

  @Override
  public String description() {
    return "Send a message to a friend";
  }

  @Override
  public String syntax() {
    return "/f msg <friend>";
  }

  @Override
  public String permission() {
    return "rfriends.command.message";
  }

  @Override
  public String[] aliases() {
    return new String[]{"msg"};
  }

  @Override
  public void onCommand(ProxiedPlayer player, String[] args) {

    try {
      ProxiedPlayer recipient = ProxyServer.getInstance().getPlayer(args[1]);
/*      if (!getDatabase().isAlreadyFriend(player, recipient)) {
        player.sendMessage(Utils.format(getPlugin().getMessagesConfig().getConfig().getString("error.not-friends")));
        return;
      }*/
      StringBuilder message = new StringBuilder();
      for (int i = 2; i < args.length; i++) {
        message.append(args[i]).append(" ");
      }
      getDatabase().sendMessage(player, recipient, message.toString());
    }catch (IndexOutOfBoundsException ex) {
      player.sendMessage(Utils.format(getPlugin().getMessagesConfig().getConfig().getString("error.invalid-arguments").replaceAll("<syntax>", syntax())));
    }
  }
}

File: RFriends-spigot/src/main/java/ga/justreddy/wiki/rfriendsspigot/commands/MessageCommand.kt
package ga.justreddy.wiki.rfriendsspigot.commands

import ga.justreddy.wiki.rfriendsspigot.enums.Messages
import ga.justreddy.wiki.rfriendsspigot.helpers.command.BaseCommand
import ga.justreddy.wiki.rfriendsspigot.helpers.friendHelper
import org.bukkit.Bukkit
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import wiki.justreddy.ga.reddyutils.uitl.ChatUtil

class MessageCommand : ChatUtil, BaseCommand(
    "message",
    "Message a friend",
    "/f message <player> <message>",
    "rfriends.command.message",
    true,
    "msg"
) {

    override fun onCommand(player: Player, args: Array<String>) {
        try{
            val friend = Bukkit.getPlayer(args[1])!!
            val builder: StringBuilder = StringBuilder()
            for (i in 2 until args.size) {
                builder.append(args[i]).append(" ")
            }

            val msg = builder.toString()
            friendHelper.sendMessage(player, friend, msg)


        }catch (e: IndexOutOfBoundsException) {
            player.sendMessage(Messages.INVALID_ARGUMENTS.toString().replace("%syntax%", getSyntax()))
        }

    }

    override fun onCommand(sender: CommandSender, args: Array<String>) {
        // Player only cmd, no need for this function
    }
}